src/analyzer/hydrate.server.ts

```diff
@@ -1,100 +1,210 @@
-import { analyzeRenamedFile, lookupFileInTree } from "./util.server"
+import { analyzeRenamedFile } from "./util.server"
 import { GitCaller } from "./git-caller.server"
 import { getCoAuthors } from "./coauthors.server"
 import { log } from "./log.server"
 import { gitLogRegex, contribRegex } from "./constants"
-
-const renamedFiles = new Map<string, string>()
-
-export async function hydrateData(
-  commit: GitCommitObject
-): Promise<[HydratedGitCommitObject, string[], Record<string, GitLogEntry>]> {
-  const authors = new Set<string>()
-  const data = commit as HydratedGitCommitObject
-  const commits: Record<string, GitLogEntry> = {}
-
-  initially_mut(data)
-
-  const commitCount = await GitCaller.getInstance().getCommitCount()
-  const commitsPerRun = 20000
-  for (let commitIndex = 0; commitIndex < commitCount; commitIndex += commitsPerRun) {
-    const gitLogResult = await GitCaller.getInstance().gitLog(commitIndex, commitsPerRun)
-
-    const matches = gitLogResult.matchAll(gitLogRegex)
-    for (const match of matches) {
-      const groups = match.groups ?? {}
-      const author = groups.author
-      const time = Number(groups.date)
-      const body = groups.body
-      const message = groups.message
-      const hash = groups.hash
-      commits[hash] = { author, time, body, message, hash }
-      const contributionsString = groups.contributions
-      const coauthors = body ? getCoAuthors(body) : []
-
-      log.debug(`Checking commit from ${time}`)
-
+import { cpus } from "os"
+
+let renamedFiles: Map<string, {path: string, timestamp: number}[]>
+let authors: Set<string>
+
+export function gatherCommitsFromGitLog(gitLogResult: string, commits: Map<string, GitLogEntry>, handleAuthors: boolean) {
+  const matches = gitLogResult.matchAll(gitLogRegex)
+  for (const match of matches) {
+    const groups = match.groups ?? {}
+    const author = groups.author
+    const time = Number(groups.date)
+    const body = groups.body
+    const message = groups.message
+    const hash = groups.hash
+    const contributionsString = groups.contributions
+    const coauthors = body ? getCoAuthors(body) : []
+    const fileChanges: FileChange[] = []
+    
+    log.debug(`Checking commit from ${time}`)
+
+    if (handleAuthors) {
       authors.add(author)
-      coauthors.forEach((coauthor) => authors.add(coauthor.name))
+      for (const coauthor of coauthors) authors.add(coauthor.name)
+    }
 
-      if (!contributionsString) continue
+    if (contributionsString) {
       const contribMatches = contributionsString.matchAll(contribRegex)
       for (const contribMatch of contribMatches) {
         const file = contribMatch.groups?.file.trim()
         const isBinary = contribMatch.groups?.insertions === "-"
         if (!file) throw Error("file not found")
 
-        const hasBeenMoved = file.includes("=>")
 
         let filePath = file
-        if (hasBeenMoved) {
-          filePath = analyzeRenamedFile(filePath, renamedFiles)
```

Comment:
I think `hasBeenMoved` was more descriptive. Could we add a comment or add the variable back?